### PR TITLE
SISRP-22622 - Switch Holds API to Pull from Hub Pt.1

### PR DESCRIPTION
**Pull request by davidlimcheng**
https://jira.berkeley.edu/browse/SISRP-22622

* Pt. 1 - Switching from Service Indicators API to Student/Academic-Status for Holds
* Service Indicators API included both positive and negative service indicators.  Holds are categorized as negative SI only, necessitating this change.
* Pt. 2 - Removing all Service Indicators API code.

### CODE REVIEW

File: app/models/my_academics/class_enrollments.rb [hunk 1]
```diff
@@ -124,6 +122,13 @@ def flatten_plan(plan)
       flat_plan
     end
 
+    def get_academic_status
+      get_academic_status = Proc.new do
+        HubEdos::AcademicStatus.new({user_id: @uid}).get
+      end
+      @academic_status ||= get_academic_status.call
+    end
```
Comment (paulkerschen) [inline app/models/my_academics/class_enrollments.rb]: @davidlimcheng A Proc can be a thing of beauty, but is there any reason this method couldn't be a one-liner? ` @academic_status ||= HubEdos::AcademicStatus.new(user_id: @uid).get`

Comment (davidlimcheng) [inline app/models/my_academics/class_enrollments.rb]: @paulkerschen I don't see why not!  Will amend.

Comment (davidlimcheng) [inline app/models/my_academics/class_enrollments.rb]: @paulkerschen Amended as a one-liner.

File: src/assets/templates/widgets/holds.html [hunk 2]
```diff
@@ -1,57 +1,54 @@
...
           >
             <td>
               <i class="fa fa-exclamation-circle cc-icon-red"></i>
-              <span data-ng-bind="indicator.reasonDescr"></span>
+              <span data-ng-bind="hold.reason.description"></span>
             </td>
-            <td data-ng-if="indicator.startDate.epoch" data-ng-bind="indicator.startDate.epoch * 1000 | date:'MM/dd/yy'" class="cc-table-right"></td>
-            <td data-ng-if="!indicator.startDate.epoch" data-ng-bind="indicator.startTermDescr" class="cc-table-right"></td>
+            <td data-ng-if="hold.fromDate" data-ng-bind="hold.fromDate | date:'MM/dd/yy'" class="cc-table-right"></td>
+            <td data-ng-if="!hold.fromDate && (hold.fromTerm && hold.fromTerm.name)" data-ng-bind="hold.fromTerm.name" class="cc-table-right"></td>
```
Comment (paulkerschen) [inline src/assets/templates/widgets/holds.html]: @davidlimcheng FYI, Angular templates are forgiving to undefined and null, to save you space on null checks. `hold.fromTerm.name` will politely return falsey if `hold.fromTerm` is falsey itself.

File: src/assets/templates/widgets/holds.html [hunk 3]
```diff
@@ -1,57 +1,54 @@
...
           </tr>
-          <tr data-ng-if="indicator.show">
+          <tr data-ng-if="hold.show">
             <td colspan="2">
-              <div class="cc-table-padding cc-text-pre-line" data-ng-if="indicator.instructions" data-ng-bind="indicator.instructions"></div>
+              <div class="cc-table-padding cc-text-pre-line" data-ng-if="hold.reason && hold.reason.formalDescription" data-ng-bind="hold.reason.formalDescription"></div>
```
Comment (paulkerschen) [inline src/assets/templates/widgets/holds.html]: @davidlimcheng ditto

Comment (davidlimcheng) [inline src/assets/templates/widgets/holds.html]: @paulkerschen Amended!

**Comment by paulkerschen** [comment]
Thanks @davidlimcheng, waiting on Travis...

**Comment by johncrossman** [comment]
Travis is happy. Thx @davidlimcheng

**Pull request merged**